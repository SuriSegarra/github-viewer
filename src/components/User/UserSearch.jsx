import React from 'react';
import PropTypes from 'prop-types';

const userSearch = ({ username, onUserChange, onUserSubmit }) => (
  <>
    <input type='text' name='username' value={username} onChange={onUserChange}/>
    <input type='button' name='search' value='Search' onClick={onUserSubmit}></input>
  </>
);

userSearch.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};

export default userSearch;
