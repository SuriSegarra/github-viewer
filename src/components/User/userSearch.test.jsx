import React from 'react';
import { shallow } from 'enzyme';

import UserSearch from './UserSearch';

describe('UserSearch', () => {
  it('matches a userSearch snapshot', () => {
    const wrapper = shallow(<UserSearch 
      username='Surelis Segarra'
      onUserChange={() => {}}
      onUserSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
