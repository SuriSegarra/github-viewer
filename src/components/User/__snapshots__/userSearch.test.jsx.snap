// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch matches a userSearch snapshot 1`] = `
<Fragment>
  <input
    name="username"
    onChange={[Function]}
    type="text"
    value="Surelis Segarra"
  />
  <input
    name="search"
    onClick={[Function]}
    type="button"
    value="Search"
  />
</Fragment>
`;
